### PR TITLE
feat: kubectl exec support

### LINKED ISSUE #1689: Browser version - result is displayed as column header when using grep
Screen Shot 2019-06-13 at 9 25 44 AM

### LINKED ISSUE #2117: broken support for kubectl exec -it mypod bash
follow-on from #1692

@showeimer

### PR DESCRIPTION
**Pull request by ruanxianzhi**
#### Description of what you did:
Redirect `kubectl exec` commands to `!` and make `!` for internal plugin-to-plugin use only
Fixes #2117 
Fixes #1689 

#### My PR is a:

- [ ] 💥 Breaking change
- [x] 🐛 Bug fix
- [ ] 💅 Enhancement
- [x] 🚀 New feature

#### Please confirm that your PR fulfills these requirements

- [x] Multiple commits are squashed into one commit.
- [x] The commit message follows [Conventional Commits](https://github.com/IBM/kui/blob/master/CONTRIBUTING.md#conventional-commits), which allows us to autogenerate release notes; e.g. `fix(plugins/plugin-k8s): fixed annoying bugs`
- [x] All npm dependencies are pinned.

### CODE REVIEW

File: plugins/plugin-bash-like/src/lib/cmds/catchall.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export const dispatchToShell = async ({
 }: EvaluatorArgs) => {
   debug('handling catchall', command)
 
+  if (!execOptions.exec) {
```
Comment (starpit) [inline plugins/plugin-bash-like/src/lib/cmds/catchall.ts]: i think this should be execOptions.type === ExecType.TopLevel; i.e. reject user's trying to issue `!`.

File: plugins/plugin-k8s/src/lib/controller/kubectl.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ interface KubeExecOptions extends ExecOptions {
  */
 const kubelike = /kubectl|oc/
 const isKubeLike = (command: string): boolean => kubelike.test(command)
-
+const sendToPTY = (argv: string[]): boolean => argv.length > 1 && argv[1] === 'exec'
```
Comment (starpit) [inline plugins/plugin-k8s/src/lib/controller/kubectl.ts]: separate out with whitespace, and add a comment for this function

File: plugins/plugin-bash-like/src/lib/cmds/catchall.ts (hunk 1 above)
Comment (starpit) [reply plugins/plugin-bash-like/src/lib/cmds/catchall.ts]: and it's ok not to include this change in this PR.

**Comment by starpit** [comment]
tests for kubectl exec?

File: plugins/plugin-k8s/src/lib/controller/kubectl.ts (hunk 2 above)
Comment (ydarias) [reply plugins/plugin-k8s/src/lib/controller/kubectl.ts]: I agree with the new line separation between functions, but why do we need to add a comment? Maybe calling it shouldSendToPTY is enough to understand what it does.

**Comment by starpit** [comment]
covered by #2250

**Pull request closed without merging**